Lib/asyncio/unix_events.py

```diff
@@ -305,6 +306,78 @@ def _child_watcher_callback(self, pid, returncode, transp):
                             ssl_handshake_timeout=ssl_handshake_timeout)
         return server
 
+    async def sock_sendfile(self, sock, file, offset=0, count=None):
+        if self._debug and sock.gettimeout() != 0:
+            raise ValueError("the socket must be non-blocking")
+        try:
+            os.sendfile
+        except AttributeError as exc:
+            raise NotImplementedError("os.sendfile() in not available")
```

Comment:
Actually, I think you were right initially, Andrew, to not use the `NotImplementedError`. I think it actually means "not implemented in Python code", and shouldn't be used for signalling about missing platform capabilities.  On the other hand, adding a new exception type just for sendfile doesn't feel right either.  How about we simply raise a `RuntimeError("os.sendfile() in not available")` here, what do you think?